meta-ros2/classes/ros_ament_python.bbclass

```diff
@@ -1,18 +1,34 @@
 # Copyright (c) 2018-2019 LG Electronics, Inc.
+# Copyright (c) Qualcomm Innovation Center, Inc. All rights reserved
 
 inherit setuptools3_legacy
 
 do_install:append() {
-    mkdir -p ${D}${datadir}/ament_index/resource_index/packages
-    touch ${D}${datadir}/ament_index/resource_index/packages/${ROS_BPN}
-    if test -e ${D}${libdir}/${ROS_BPN}; then
-        for i in ${D}${libdir}/${ROS_BPN}/* ${D}${libdir}/${ROS_BPN}/*/*; do
+    mkdir -p ${D}${ros_datadir}/ament_index/resource_index/packages
+    touch ${D}${ros_datadir}/ament_index/resource_index/packages/${ROS_BPN}
+    if test -e ${D}${ros_libdir}/${ROS_BPN}; then
+        for i in ${D}${ros_libdir}/${ROS_BPN}/* ${D}${ros_libdir}/${ROS_BPN}/*/*; do
             [ -f $i ] && sed -i '1c#!/usr/bin/python3' $i
         done
         true
     fi
+
+    # similar to what setuptools3_legacy.bbclass does here:
+    for i in ${D}${ros_bindir}/* ${D}${ros_sbindir}/*; do
+        if [ -f "$i" ]; then
+            sed -i -e s:${PYTHON}:${USRBINPATH}/env\ ${SETUPTOOLS_PYTHON}:g $i
+            sed -i -e s:${STAGING_BINDIR_NATIVE}:${bindir}:g $i
+        fi
+    done
 }
 
 FILES:${PN}:prepend = " \
-    ${datadir}/ament_index \
+    ${ros_datadir}/ament_index \
 "
+
+PYTHON_SITEPACKAGES_DIR = "${ros_libdir}/${PYTHON_DIR}/site-packages"
```

Comment:
Should this perhaps append to the existing variable in case any of the ROS Python applications or modules need to use the Python modules provided by the system?

Reply:
Append is not required. It's really easy to get confused. PYTHON_SITEPACKAGES_DIR is a tmp **one-valued** variable defined and used in POKY.  And there's an environment python variable PYTHONPATH which can be set to multiple values.

Definition:
https://github.com/yoctoproject/poky/blob/kirkstone/meta/classes/python3-dir.bbclass 
Assignment:
https://github.com/yoctoproject/poky/blob/fa9689923ff4b4496180886f42813e1e7be3b321/meta/classes-recipe/setuptools3_legacy.bbclass#L52

Reply:
Ah, yes.  You are correct.  Given that the purpose of PYTHON_SITEPACKAGES_DIR is mostly to direct where to install the module and that this is scoped to recipes that inherit ros_ament_python: this looks correct.

Thanks for the explanation.